src/main/java/com/bootcamp/demo/data_access/AccountRepository.java

```diff
@@ -29,14 +30,49 @@ public String add(Account account) throws ExecutionException, InterruptedExcepti
         return futureTransaction.get();
     }
 
+    @Override
+    public Account findOne(String username) throws ExecutionException, InterruptedException {
+        CollectionReference collectionReference = this.getAccountsCollection();
+        DocumentReference documentReference = collectionReference.document(username);
+        ApiFuture<DocumentSnapshot> future = documentReference.get();
+        DocumentSnapshot documentSnapshot = future.get();
+
+        if (documentSnapshot.exists()) {
+            Map<String, Object> data = documentSnapshot.getData();
+
+            String name = data.get("name").toString();
+            String email =  data.get("email").toString();
+            String phoneNumber =  data.get("phoneNumber").toString();
+            String address =  data.get("address").toString();
+            String dateOfBirth =  data.get("dateOfBirth").toString();
+
+            return new Account(name, email, username, null, phoneNumber, address, dateOfBirth);
+        }
+        return null;
+    }
+
+    @Override
+    public Account update(Account account) throws ExecutionException, InterruptedException {
+        CollectionReference collectionReference = this.getAccountsCollection();
+
+        DocumentReference documentReference = collectionReference.document(account.getUsername());
+        documentReference.set(account).get();
+
+        return this.findOne(account.getUsername());
```

Comment:
Why are you searching for and returning the updated account? The method should only be responsible for the update operation itself. Wouldn't it be better to return a string to signal that the operation was successful?

Reply:
Maybe you want to update only an attribute or two, not the whole account. So I thought it would be helpful to return the updated account so the changes made can be visible.

Reply:
We can just retrieve it after the update, I don't think we should assign more responsability than needed to the update.